Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -528,5 +528,15 @@ public func collapse<Node: SyntaxProtocol>(
     break
   }
 
-  return expansions.joined(separator: separator)
+  // Join the expansions ensuring `separator` between them.
+  var collapsed = ""
+  for expansion in expansions {
+    if (collapsed.isEmpty || expansion.hasPrefix(separator)) {
```

Comment:
Nitpick, sorry for not noticing earlier: The parentheses are not needed.

```suggestion
    if collapsed.isEmpty || expansion.hasPrefix(separator) {
```